amandaguan-ag/todo
# Design

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓  | :art: UI Redesign          |
|  ✓  | :wheelchair: Accessibility |
|  ✓  | :link: Update dependencies |

## Acceptance Criteria

- [x] Custom theme applied across the entire application.
- [x] Header component added at the top of every page.
- [x] All accessibility tests passed according to WCAG guidelines.
- [x] Design is responsive

## Description 

This PR overhauls the app's design, enhancing usability and aesthetics while ensuring accessibility standards are met.

### Apply Custom Theme

The new custom theme has been applied which aligns with our brand's visual identity. This includes changes to the color palette, typography, and component styling across the application.

### Add Header Component

A new `Header` component has been introduced, which includes navigation links and improves the overall user experience by making it easier to navigate through the app.

### Pass Accessibility Test

We have conducted thorough accessibility tests using tools like Axe and Lighthouse, ensuring our application meets the WCAG 2.1 AA standards. This includes keyboard navigation, screen reader support, and visual adjustments for color contrast.

## UI change

### Before
![image](https://github.com/amandaguan-ag/todo/assets/121919044/e48474dd-e593-4635-a5dd-314590001b77)

### After
![image](https://github.com/amandaguan-ag/todo/assets/121919044/f08f0d22-88b6-4c29-9757-bd5d49ced109)

## Testing Steps / QA Criteria

To prepare for testing, please ensure you've followed the [setup steps](https://github.com/amandaguan-ag/todo#setup) outlined in the project's README, including running any necessary database migrations to update your local database schema.

1. Checkout to this branch with `git checkout design`.
2. Make sure you are in the backend directory and run the database migrations to update your schema by executing `npm run typeorm migration:run`.
3. Start the backend service by running `npm run start` in the backend directory.
4. In a separate terminal, navigate to the frontend directory and run `npm start` to launch the frontend.
5. Navigate to [http://localhost:3000](http://localhost:3000) to access the Todo List Application.
* Navigate through the app and observe the new header on every page.
* Use accessibility tools to verify that the app meets accessibility standards.

## TODOs

- UI consistency checks across different devices and browsers have been scheduled.
- Further accessibility enhancements are under review to ensure compliance with upcoming regulations.
